src/app/(tools)/coverLetter/page.tsx

```diff
@@ -0,0 +1,285 @@
+"use client";
+
+import { useState, useRef } from 'react';
+import { Upload, X } from 'lucide-react';
+import axios from 'axios';
+import jsPDF from 'jspdf';
+import { useForm, Controller } from 'react-hook-form';
+import { z } from 'zod';
+import { zodResolver } from '@hookform/resolvers/zod';
+
+import { Button } from "@/components/ui/button";
+import { Textarea } from "@/components/ui/textarea";
+import { toast } from '@/hooks/use-toast';
+
+const schema = z.object({
```

Comment:
schema, FormData  are generic names. 

use specific names for variables based on the intended purpose 

maybe coverLetterInputSchema etc 


also these could have moved to an new file. for all types and validations